### PR TITLE
Selfie master

### PR DESCRIPTION
**Pull request by sigsegv4**
Asg0

### CODE REVIEW

File: selfie.c [hunk 1]
```diff
@@ -566,6 +566,97 @@ initParser() {
...
+// | Entry         |
+// +---------------+
+// | Entry 2       |
+// +---------------+
+// | Pr next 0     |
+// +---------------+
+// | Pr Prev 0     |
+// +---------------+
+   
+int createLList(int size){
```
Comment (maigner) [inline selfie.c]: why not int\* createLList... ? It returns a pointer, no? Same with the other operations in your list.

File: selfie.c [hunk 2]
```diff
@@ -566,6 +566,97 @@ initParser() {
...
+// ---------------------     O S   ---------------------------------
+// -----------------------------------------------------------------
+// *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~
+
+
+// -----------------------------------------------------------------
+// ------------------------- Assignment 0 Linked List --------------
+// -----------------------------------------------------------------
+
+// Eg.: Size 2
```
Comment (maigner) [inline selfie.c]: Think about the following data layout: Place the previous and next pointer at the beginning of the node. Then, the size of the node does not matter

+// +---------------+
+// | prev         |
+// +---------------+
+// | next         |
+// +---------------+
+// | payload starts here |
+// +---------------+
+// | and so on     |
+// +---------------+

File: selfie.c [hunk 3]
```diff
@@ -566,6 +566,97 @@ initParser() {
...
+   list = malloc((size + 2) * 4);
+   pr = list;
+   pr = pr + size;
+   *pr = 0;
+   pr = pr + 1;
+   *pr = 0;
+   return list;
+}
+
+int addNodeToLList(int size, int list){
```
Comment (maigner) [inline selfie.c]: see comment on data layout. Once you've got the convention that prev and next are the first 2 words in any node "structure", you get much simpler code for insertion and removal of nodes.

**Comment by sigsegv4** [comment]
Have fixed the list, in Assg1

**Pull request closed without merging**